framework/src/main/java/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1504,6 +1516,33 @@ private Object convert(Number number, TypeKind expectedKind) {
         }
     }
 
+    /**
+     * Converts {@code number} to {@code expectedKind} * -1
+     *
+     * @param number Number value to be converted
+     * @param expectedKind converts the Number to {byte, short, int, long, float, double}
+     * @return the converted Number Object
+     */
+    private Object convertNegate(Number number, TypeKind expectedKind) {
```

Comment:
It's unfortunate to have all this code duplication.
I would have created a version with a boolean `negate`.  Then the first line could set
```
scalefactor = negate ? -1 : 1;
```
and the body `convert` could be a call to that method with the boolean false.
The extra code duplication, while unfortunate, isn't a huge deal, though.

Reply:
I could definitely add another parameter to the `convert` method but I don't think Java handles default parameter values as nicely as other languages like C++. I could add another parameter and change the places where `convert` is called or I can add a vararg but that will look weird with wanting only one parameter. Which one do you prefer?